Fix ruler storage single binary mode

pracucci opened the pull request:
**What this PR does**:
In #3805 we've introduced `-ruler-storage.*` and then we depreacted the old `-ruler.storage`.

We have a check in `initRulerStorage()` to not start the ruler at all when Cortex has been started in single-binary mode and the ruler hasn't been configured. However, we forgot to add a check in the new ruler storage config, which should be fixed by this PR.

**Which issue(s) this PR fixes**:
Fixes #4239

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

pstibrany reviewed (approved):
Makes sense.

pstibrany commented:
I'm wondering if we should add a test to main cortex package for this scenario? That should be pretty easy.

gotjosh commented on CHANGELOG.md:
```diff
@@ -43,6 +43,7 @@
 * [BUGFIX] Ruler: fix `/ruler/rule_groups` endpoint doesn't work when used with object store. #4182
 * [BUGFIX] Ruler: Honor the evaluation delay for the `ALERTS` and `ALERTS_FOR_STATE` series. #4227
 * [BUGFIX] Fixed cache fetch error on Redis Cluster. #4056
+* [BUGFIX] Ruler: fix ruler startup in single-binary mode when the new `ruler_storage` is used. #4252
```
```suggestion
* [BUGFIX] Ruler: fix startup in single-binary mode when the new `ruler_storage` is used. #4252
```

pracucci commented:
> I'm wondering if we should add a test to main cortex package for this scenario? That should be pretty easy.

Good point. I did two things:
1. Added a unit test on `initRulerStorage()` (as discussed offline)
2. Switched the getting started config to use the new `ruler_storage` (the old one is deprecated now) so that the `TestRulerAPISingleBinary` and `TestRulerEvaluationDelay` integration tests run with that

@pstibrany WDYT?

pstibrany commented:
> Good point. I did two things:
> 
> 1. Added a unit test on `initRulerStorage()` (as discussed offline)
> 2. Switched the getting started config to use the new `ruler_storage` (the old one is deprecated now) so that the `TestRulerAPISingleBinary` and `TestRulerEvaluationDelay` integration tests run with that
> 
> @pstibrany WDYT?

Looks great, thank you very much!
